datasource/src/test/scala/quasar/plugin/mariadb/datasource/MariaDbDatasourceSpec.scala

```diff
@@ -0,0 +1,308 @@
+              "med" -> rLong(-8388608L),
+              "norm" -> rLong(-2147483648L),
+              "big" -> rLong(-9223372036854775808L),
+              "dbl" -> rDouble(-1.7976931348623157E+308),
+              "flt" -> rDouble(-3.40282E+38),
+              "xct" -> rNum(BigDecimal("-99999999999999999999999999999999999.9999999999999999999999999999"))),
+            obj(
+              "tiny" -> rLong(127L),
+              "small" -> rLong(32767L),
+              "med" -> rLong(8388607L),
+              "norm" -> rLong(2147483647L),
+              "big" -> rLong(9223372036854775807L),
+              "dbl" -> rDouble(1.7976931348623157E+308),
+              "flt" -> rDouble(3.40282E+38),
+              "xct" -> rNum(BigDecimal("99999999999999999999999999999999999.9999999999999999999999999999"))))
+
+          results must containTheSameElementsAs(expected)
+        }
+      }
+    }
+
+    // TIME | DATE | TIMESTAMP | DATETIME | YEAR
+    "temporal" >> {
+      val minDate = LocalDate.parse("1000-01-01")
+      val maxDate = LocalDate.parse("9999-12-31")
+
+      val minTime = LocalTime.parse("00:00:00")
+      val maxTime = LocalTime.parse("23:59:59.999999")
+
+      val minDateTime = LocalDateTime.parse("1000-01-01T00:00:00.000000")
+      val maxDateTime = LocalDateTime.parse("9999-12-31T23:59:59.999999")
+
+      val minTimestamp = LocalDateTime.parse("1970-01-01T00:00:01")
+      val maxTimestamp = LocalDateTime.parse("2038-01-19T03:14:07")
+
+      val minYear = 1901
+      val maxYear = 2155
+
+      def insert(tbl: String, lt: LocalTime, ld: LocalDate, ts: LocalDateTime, dt: LocalDateTime, y: Int): ConnectionIO[Int] = {
+        val sql =
+          fr"INSERT INTO" ++ frag(tbl) ++ fr0" (lt, ld, ts, dt, y) values ($lt, $ld, $ts, $dt, $y)"
+
+        sql.update.run
+      }
+
+      harnessed() use { case (xa, ds, path, name) =>
+        val setup = for {
+          _ <- (fr"CREATE TABLE" ++ frag(name) ++ fr0" (lt TIME(6), ld DATE, ts TIMESTAMP(6), dt DATETIME(6), y YEAR)").update.run
+
+          _ <- insert(name, minTime, minDate, minTimestamp, minDateTime, minYear)
+          _ <- insert(name, maxTime, maxDate, maxTimestamp, maxDateTime, maxYear)
+        } yield ()
+
+        (setup.transact(xa) >> loadRValues(ds, path)) map { results =>
+          val expected = List(
+            obj(
+              "lt" -> rLocalTime(minTime),
+              "ld" -> rLocalDate(minDate),
+              "ts" -> rLocalDateTime(minTimestamp),
+              "dt" -> rLocalDateTime(minDateTime),
+              "y" -> rLong(minYear.toLong)),
+            obj(
+              "lt" -> rLocalTime(maxTime),
+              "ld" -> rLocalDate(maxDate),
+              "ts" -> rLocalDateTime(maxTimestamp),
+              "dt" -> rLocalDateTime(maxDateTime),
+              "y" -> rLong(maxYear.toLong)))
+
+          results must containTheSameElementsAs(expected)
+        }
+      }
+    }
+/* TODO: Only supported by mariadb
```

Comment:
Should this TODO be taken care of now, or is there a reason it can't be?

Reply:
The only reason would be timing, the `TODO` is just for the test as the suite currently doesn't support selecting tests based on vendor. I discovered this last-minute and figured we could enhance things in a future PR.

Reply:
I don't mind adding it if we think it'd be better to have sooner than later.

Reply:
It actually turned out to be pretty easy to add, fixed 👍